nntrainer/layers/rnn.cpp

```diff
@@ -163,11 +164,68 @@ void RNNLayer::copy(std::shared_ptr<Layer> l) {
 }
 
 void RNNLayer::calcDerivative() {
-  //  NYI
+  Tensor &derivative_ = net_hidden[0]->getGradientRef();
+  Tensor &weight =
+    weightAt(static_cast<int>(RNNParams::weight_xh)).getVariableRef();
+  Tensor &ret_ = net_input[0]->getGradientRef();
+
+  ret_ = derivative_.dot(weight, ret_, false, true);
```

Comment:
Minor but this removes extra copy.

```suggestion
  derivative_.dot(weight, ret_, false, true);
```

Reply:
done.